src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/DateFormatIT.java

```diff
@@ -135,6 +137,29 @@ public void sortByDateFormat() throws IOException {
                 is(new DateTime("2014-08-24T07:00:55.481Z", DateTimeZone.UTC)));
     }
 
+    @Test
+    public void groupByAndSort() throws IOException {
+        JSONObject aggregations = executeQuery(
+                "SELECT date_format(insert_time, 'dd-MM-YYYY') " +
+                        "FROM elasticsearch-sql_test_index_online " +
+                        "GROUP BY date_format(insert_time, 'dd-MM-YYYY') " +
+                        "ORDER BY date_format(insert_time, 'dd-MM-YYYY') DESC")
+                .getJSONObject("aggregations");
+
+        String date = DateFormatTest.getScriptAggregationKey(aggregations, "date_format");
+        JSONArray buckets = aggregations.getJSONObject(date).getJSONArray("buckets");
+
+        assertThat(buckets.length(), is(8));
+
+        String previousKey = buckets.getJSONObject(0).getString("key");
+        for (int i=1; i < buckets.length(); i++) {
```

Comment:
nit: space between `=`. And just a proposal, a custom Hamcrest `Matcher` to check collection is sorted could provide higher abstraction and be helpful to many test cases regarding `ORDER BY`.

Reply:
Fixed formatting.
Was able to have oneliner without the need to introduce custom matcher